### PR TITLE
Add AuthCodeNonce for OAuth 2.0 OC Communication

### PR DESCRIPTION
**Pull request by charliewang95**
Changes: 

1. Put nonce value in configuration (with default) which will be used when sending auth requests
2. Update said nonce if we receive AuthCodeNonce header in response (sent by ChannelConnector)
3. Pass a callback function to axiosRetry so that retries are also honoring the nonce update logic

### CODE REVIEW

File: src/SDK.ts [hunk 1]
```diff
@@ -47,24 +47,25 @@ import sessionInitRetryHandler from "./Utils/SessionInitRetryHandler";
...
+    getReconnectAvailability: 10000,
     submitPostChatResponse: 10000,
     getSurveyInviteLink: 10000,
     getChatTranscripts: 30000,
     emailTranscript: 5000,
-    fetchDataMaskingInfo: 1000,
-    makeSecondaryChannelEventRequest: 5000,
-    getAgentAvailability: 5000,
+    fetchDataMaskingInfo: 5000,
+    makeSecondaryChannelEventRequest: 10000,
```
Comment (josuhern) [inline src/SDK.ts:61]: Why are we incrementing the default timeout?

**Review by josuhern — APPROVED** [review]

File: src/SDK.ts [hunk 2]
```diff
@@ -47,24 +47,25 @@ import sessionInitRetryHandler from "./Utils/SessionInitRetryHandler";
...
     getChatConfig: 30000,
-    getLWIDetails: 5000,
+    getLWIDetails: 10000,
     getChatToken: 10000,
-    sessionInit: 5000,
+    sessionInit: 10000,
     sessionClose: 10000,
-    getReconnectableChats: 5000,
-    getReconnectAvailability: 5000,
+    getReconnectableChats: 10000,
```
Comment (josuhern) [inline src/SDK.ts:54]: Why are we incrementing the default timeout?

File: src/SDK.ts [hunk 3]
```diff
@@ -795,7 +845,11 @@ export default class SDK implements ISDK {
     return new Promise(async (resolve, reject) => {
       try {
         const response = await axiosInstance(options);
-        const { data } = response;
+        const { data, headers } = response;
+        if (headers?.authcodenonce) {
+          this.configuration.authCodeNonce = headers?.authcodenonce;
+        }
```
Comment (elopezanaya) [inline src/SDK.ts:851]: my suggestion is to abstract this to a setAtuhCodeNonce(headers) function , instead of repeat it 10 times ,

**Review by elopezanaya — APPROVED** [review]

File: src/SDK.ts (hunk 2 above)
Comment (charliewang95) [reply src/SDK.ts:54]: With OAuth 2.0 ChannelConnector needs more time to process, and 5 seconds is sometime not enough. Before backend improves its performance, we must increase the timeout here

File: src/SDK.ts (hunk 3 above)
Comment (charliewang95) [reply src/SDK.ts]: done

**Pull request merged**